src/overlays/actors/ovl_Bg_Tobira01/z_bg_tobira01.c

```diff
@@ -28,14 +28,79 @@ const ActorInit Bg_Tobira01_InitVars = {
     (ActorFunc)BgTobira01_Update,
     (ActorFunc)BgTobira01_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Tobira01/func_80B12430.s")
+void BgTobira01_Action(BgTobira01* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+    s16 csId = this->dyna.actor.csId;
+    s16 tempPosY;
+
+    if (this->cutsceneRequested) {
+        if (CutsceneManager_GetCurrentCsId() == CS_ID_GLOBAL_TALK) {
+            CutsceneManager_Stop(CS_ID_GLOBAL_TALK);
+        } else if (CutsceneManager_IsNext(csId)) {
+            CutsceneManager_StartWithPlayerCs(csId, &this->dyna.actor);
+            SET_WEEKEVENTREG(WEEKEVENTREG_88_40);
+            this->cutsceneRequested = false;
+        } else {
+            CutsceneManager_Queue(csId);
+        }
+    } else if (!CHECK_WEEKEVENTREG(WEEKEVENTREG_88_40) && (this->posYTick == 0) &&
+               (play->actorCtx.playerImpact.timer != 0) &&
+               (play->actorCtx.playerImpact.type == PLAYER_IMPACT_GORON_GROUND_POUND) &&
+               (SurfaceType_GetSceneExitIndex(&play->colCtx, player->actor.floorPoly, player->actor.floorBgId) == 6)) {
+        this->cutsceneRequested = true;
+        this->unk16C = 0;
+    }
+
+    tempPosY = this->posYTick;
+    if (CHECK_WEEKEVENTREG(WEEKEVENTREG_88_40)) {
+        this->posYTick++;
+    } else {
+        this->posYTick--;
+    }
+
+    this->posYTick = CLAMP(this->posYTick, 0, 60);
+    if (tempPosY != this->posYTick) {
+        Actor_PlaySfx(&this->dyna.actor, NA_SE_EV_STONEDOOR_OPEN_S - SFX_FLAG);
+        this->dyna.actor.world.pos.y = this->posY = (this->posYTick * (5.0f / 3.0f)) + this->dyna.actor.home.pos.y;
+        this->timer = 180;
+    }
+
+    if (!(player->stateFlags1 & PLAYER_STATE1_40) && CHECK_WEEKEVENTREG(WEEKEVENTREG_88_40) && DECR(this->timer) == 0) {
+        CLEAR_WEEKEVENTREG(WEEKEVENTREG_88_40);
+    }
+}
+
+void BgTobira01_Init(Actor* thisx, PlayState* play) {
+    BgTobira01* this = THIS;
+
+    DynaPolyActor_Init(&this->dyna, 1);
```

Comment:
```suggestion
    DynaPolyActor_Init(&this->dyna, DYNA_TRANSFORM_POS);
```
This takes an enum arg